fdbserver/Status.actor.cpp

```diff
@@ -2375,8 +2388,8 @@ static JsonBuilderObject faultToleranceStatusFetcher(DatabaseConfiguration confi
 	// max zone failures that we can tolerate to not lose data
 	int zoneFailuresWithoutLosingData = std::min(maxZoneFailures, maxCoordinatorZoneFailures);
 
-	if (minReplicasRemaining >= 0) {
-		zoneFailuresWithoutLosingData = std::min(zoneFailuresWithoutLosingData, minReplicasRemaining - 1);
+	if (minStorageReplicasRemaining >= 0) {
+		zoneFailuresWithoutLosingData = std::min(zoneFailuresWithoutLosingData, minStorageReplicasRemaining - 1);
 	}
```

Comment:
I don't quite understand this if statement.

If we don't have any storage replica remaining, won't we not be able to tolerate any failure at all? If so, why not updating zoneFailuresWithoutLosingData when minStorageReplicasRemaining == 0?

Reply:
This is saying you can lose all but one of your replicas without losing data. For example if minStorageReplicasRemaining is 2, we want the tolerated failures to be 1.

The `if` is protecting against the case where we couldn't read the number of replicas remaining, in which case the value will be -1.

Reply:
Thanks! That makes sense now. I think I overlooked the "="....